modules/navier_stokes/src/fvbcs/INSFVMomentumAdvectionOutflowBC.C

```diff
@@ -46,6 +46,27 @@ INSFVMomentumAdvectionOutflowBC::INSFVMomentumAdvectionOutflowBC(const InputPara
     mooseError("In threedimensions, the w velocity must be supplied using the 'w' parameter");
 }
 
+ADReal
+INSFVMomentumAdvectionOutflowBC::computeSegregatedContribution()
```

Comment:
The code that follows is the core of the description of this object. And it's repeated in `gatherRCData`. When someone in the future might want to modify the core of this object, it would be ideal for them to only have to do it in one place. Have you considered whether this can be easily done?

Reply:
I have, but the fact that we are taking the derivatives of the variable on the face makes the generalization it non-trivial. I'll push another version in am inute.